src/editor/lexercpp.cpp

```diff
@@ -1,6 +1,212 @@
 #include "lexercpp.h"
+#include <QDebug>
 
 LexerCPP::LexerCPP()
 {
 
 }
+
+LexerCPP::~LexerCPP()
+{
+
+}
+
+bool LexerCPP::isKeyword(QString lexem)
+{
+    return KEYWORDS.contains(lexem);
+}
+
+bool LexerCPP::isIdentifier(QString lexem)
+{
+    QRegExp rx("[A-Za-z_][A-Za-z0-9_]*");
+    return rx.exactMatch(lexem);
+}
+
+bool LexerCPP::isNumber(QString lexem)
+{
+    QRegExp rx("[0-9]+");
+    return rx.exactMatch(lexem);
+}
+
+bool LexerCPP::isFloatNumber(QString lexem)
+{
+    QRegExp rx("[0-9]+\\.[0-9]+");
```

Comment:
All these functions have the same code. It can be one function with an additional parameter.